lib/device_api/android/adb.rb

```diff
@@ -233,6 +233,34 @@ def self.screencap( serial, args )
         shell(serial, cmd)
       end
 
+      def self.check_ip_address(ipaddressandport)
+        unless ipaddressandport =~ /\A(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3}):[0-9]+\Z/ 
+             raise ADBCommandError.new("Invalid IP address and port " + ipaddressandport)
+        end
+      end
+
+      def self.connect(ipaddressandport)
+        check_ip_address(ipaddressandport)
+        cmd = "adb connect " + ipaddressandport
+        result = execute(cmd)
+        if result.stdout.to_s =~ /.*already connected to.*/
+          raise DeviceAlreadyConnectedError.new("Device " + ipaddressandport + " already connected")
+        elsif result.stdout.to_s =~ /.*connected to.*/
+          return
+        else 
+          raise ADBCommandError.new("Unable to adb connect to " + ipaddressandport + " result was: " + result.stdout.inspect)
+        end 
+      end
+
+      def self.disconnect(ipaddressandport)
```

Comment:
Like `self.disconnect` the port should be optional, and possibly not required at all.